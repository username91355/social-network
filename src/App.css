.app_wrapper {
    width: 1080px;
    margin: 0 auto;
    display: grid;
    grid-template-areas:
            'h h h'
            's c c'
            '. c c';
    grid-template-columns: 2fr 5fr 5fr;
    grid-template-rows: 60px 150px 100%;
    grid-gap: 10px;
}

.app_content {
    grid-area: c;
    background-color: #1E1F26;
}

/*Palitra {*/
/*#D0E1F9 - Фиолетово-голубой*/

/*#4D648D - Черника*/

/*#283655 - Чернильный индиго*/

/*#1E1F26 - Полуночныйсиний*/

/*     #F9F9FF - белый */
/*}*/