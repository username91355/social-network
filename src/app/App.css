.app__wrapper {
    max-width: 1080px;
    margin: 0 auto;
    display: grid;
    grid-template-areas:
            'h h'
            'n m'
            '. m'
            'f f';
    grid-template-columns: 1fr 4fr;
    grid-template-rows: 60px auto auto auto;
    grid-gap: 10px;
}

header {
    grid-area: h;
}

nav {
    grid-area: n;
    display: flex;
    flex-direction: column;
    position: sticky;
    top: 0px;
}

main {
    grid-area: m;
    min-height: 100vh;
    background-color: #001530;
}

footer {
    text-align: center;
    grid-area: f;
    padding: 30px;
    background-color: #001530;
    color: #f0f2f5;
}