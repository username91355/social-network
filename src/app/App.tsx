import React, {Suspense, useEffect} from 'react'
import styles from './App.module.css'
import {Header} from '../components/header/Header'
import {Nav} from '../components/nav/Nav'
import {Profile} from '../pages/profile/Profile'
import {Login} from '../pages/login/Login'
import {Footer} from '../components/footer/Footer'
import {PageNotFound} from '../components/page-not-found/PageNotFound'
import {TAppState} from '../state/store'
import {appInitialization, AppStatus} from '../state/reducers/app-reducer'
import {useDispatch, useSelector} from 'react-redux'
import {Navigate, Route, Routes} from 'react-router-dom'
import {Preloader} from '../components/preloader/Preloader'
import {ErrorSnackbar} from '../components/error-snackbar/ErrorSnackbar'

const Users = React.lazy(() => import('../pages/users/Users'))
const Messages = React.lazy(() => import('../pages/messages/Messages'))
const Chat = React.lazy(() => import('../pages/chat/Chat'))

export const App: React.FC = () => {

    const
        dispatch = useDispatch(),
        appStatus = useSelector((state: TAppState) => state.app.appStatus),
        appError = useSelector((state: TAppState) => state.app.error)

    useEffect(() => {
        dispatch(appInitialization())
    }, [dispatch])

    return (
        <div className={styles.app__wrapper}>
            <ErrorSnackbar error={appError}/>
            <Header/>
            <Nav/>
            <main>{(appStatus === AppStatus.IDLE || appStatus === AppStatus.LOADING)
                ? <Preloader/>
                : <Routes>
                    <Route path='/' element={<Navigate to='/profile'/>}/>
                    <Route path='/profile' element={<Profile/>}/>
                    <Route path='/profile/:userId' element={<Profile/>}/>
                    <Route path='/users' element={
                        <Suspense fallback={<Preloader/>}>
                            <Users/>
                        </Suspense>}/>
                    <Route path='/messages' element={
                        <Suspense fallback={<Preloader/>}>
                            <Messages/>
                        </Suspense>}/>
                    <Route path='/messages/:userId' element={
                        <Suspense fallback={<Preloader/>}>
                            <Messages/>
                        </Suspense>}/>
                    <Route path='/Chat' element={
                        <Suspense fallback={<Preloader/>}>
                            <Chat/>
                        </Suspense>}/>
                    <Route path='/login' element={<Login/>}/>
                    <Route path='*' element={<PageNotFound/>}/>
                </Routes>
            }</main>
            <Footer/>
        </div>
    )
}