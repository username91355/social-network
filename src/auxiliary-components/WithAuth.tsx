import React from "react";
import {TAppState} from "../state/store";
import {useSelector} from "react-redux";
import {Navigate, useLocation} from "react-router-dom";

interface IWithAuth {
    children: React.ReactNode
}

export const WithAuth = ({children}: IWithAuth) => {

    const isAuth = useSelector((state: TAppState) => state.app.isAuth)
    const location = useLocation()

    return (!isAuth)
        ? <Navigate to='/login' state={location.pathname}/>
        : <>{children}</>
}
