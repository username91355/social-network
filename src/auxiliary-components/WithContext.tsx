import React from "react";
import {BrowserRouter} from "react-router-dom";
import {Provider} from "react-redux";
import {store} from "../state/store";

interface ITestComponentWithContext {
    children: React.ReactNode
}

export const WithContext = ({children}: ITestComponentWithContext) => {
    return (
        <BrowserRouter>
            <Provider store={store}>
                {children}
            </Provider>
        </BrowserRouter>
    )
}