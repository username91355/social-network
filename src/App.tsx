import React from 'react'
import './App.css'
import Sidebar from './components/common/sidebar/Sidebar'
import Header from './components/common/header/HeaderC'
import Users from './components/users/UsersC'
import { Route, Routes } from 'react-router-dom'

function App() {

    return (
        <div className='app_wrapper'>
            <Header/>
            <Sidebar/>
            <div className='app_content'>
                <Routes>
                    <Route path='/users' element={<Users/>}/>
                </Routes>
            </div>
        </div>
    );
}

export default App;
