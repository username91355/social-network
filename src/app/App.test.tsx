import {cleanup, render, screen} from '@testing-library/react'
import {App} from './App'
import React from 'react'
import {WithContext} from '../auxiliary-components/WithContext'

// window.matchMedia = window.matchMedia || function() {
//     return {
//         matches: false,
//         addListener: jest.fn(),
//         removeListener: jest.fn()
//     }
// }

describe('App test`s', () => {
    beforeAll(() => {
        window.matchMedia = window.matchMedia || function () {
            return {
                matches: false,
                addListener: jest.fn(),
                removeListener: jest.fn()
            }
        }
    });
    afterAll(cleanup)
    // Object.defineProperty(window, "matchMedia", {
    //     writable: true,
    //     value: jest.fn().mockImplementation(query => ({
    //         matches: false,
    //         media: query,
    //         onchange: null,
    //         addListener: jest.fn(), // Deprecated
    //         removeListener: jest.fn(), // Deprecated
    //         addEventListener: jest.fn(),
    //         removeEventListener: jest.fn(),
    //         dispatchEvent: jest.fn(),
    //     }))
    // });
    it('should be render', () => {
        render(
            <WithContext>
                <App/>
            </WithContext>
        )

        expect(screen.getByText(/profile/ig)).toBeInTheDocument()
    })
    it('App snapshot', () => {
        const element = render(
            <WithContext>
                <App/>
            </WithContext>
        )

        expect(element).toMatchSnapshot()
    })
})