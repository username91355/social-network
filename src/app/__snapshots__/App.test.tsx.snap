// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test\`s App snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="app__wrapper"
      >
        <header
          class="header__wrapper"
        >
          <div>
            <div
              class="header__mobile_menu"
            >
              <button
                class="ant-btn ant-btn-primary"
                type="button"
              >
                <span
                  aria-label="menu-unfold"
                  class="anticon anticon-menu-unfold"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="menu-unfold"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
                    />
                  </svg>
                </span>
              </button>
              <div>
                Menu
              </div>
            </div>
            <img
              alt="logo"
              class="header__logo"
              src="logo.png"
            />
          </div>
          <div>
            <a
              href="/login"
            >
              Login
            </a>
          </div>
        </header>
        <nav
          class="nav__wrapper"
        >
          <ul
            class="ant-menu ant-menu-root ant-menu-inline ant-menu-light nav__menu"
            data-menu-list="true"
            role="menu"
            tabindex="0"
          >
            <li
              class="ant-menu-item ant-menu-item-only-child"
              data-menu-id="rc-menu-uuid-test-/profile"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                class="ant-menu-title-content"
              >
                <h3>
                  <a
                    class=""
                    href="/profile"
                  >
                    Profile
                  </a>
                </h3>
              </span>
            </li>
            <li
              class="ant-menu-item ant-menu-item-only-child"
              data-menu-id="rc-menu-uuid-test-/users"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                class="ant-menu-title-content"
              >
                <h3>
                  <a
                    class=""
                    href="/users"
                  >
                    Users
                  </a>
                </h3>
              </span>
            </li>
            <li
              class="ant-menu-item ant-menu-item-only-child"
              data-menu-id="rc-menu-uuid-test-/messages"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                class="ant-menu-title-content"
              >
                <h3>
                  <a
                    class=""
                    href="/messages"
                  >
                    Messages
                  </a>
                </h3>
              </span>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display: none;"
          />
        </nav>
        <main>
          <div
            class="preloader__wrapper"
          >
            <div
              class="ant-spin ant-spin-lg ant-spin-spinning"
            >
              <span
                class="ant-spin-dot ant-spin-dot-spin"
              >
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
                <i
                  class="ant-spin-dot-item"
                />
              </span>
            </div>
          </div>
        </main>
        <footer
          class="footer__wrapper"
        >
          Social network ©2021 Created by Balyaev Dmitriy with Ant Design
        </footer>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="app__wrapper"
    >
      <header
        class="header__wrapper"
      >
        <div>
          <div
            class="header__mobile_menu"
          >
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span
                aria-label="menu-unfold"
                class="anticon anticon-menu-unfold"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="menu-unfold"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM142.4 642.1L298.7 519a8.84 8.84 0 000-13.9L142.4 381.9c-5.8-4.6-14.4-.5-14.4 6.9v246.3a8.9 8.9 0 0014.4 7z"
                  />
                </svg>
              </span>
            </button>
            <div>
              Menu
            </div>
          </div>
          <img
            alt="logo"
            class="header__logo"
            src="logo.png"
          />
        </div>
        <div>
          <a
            href="/login"
          >
            Login
          </a>
        </div>
      </header>
      <nav
        class="nav__wrapper"
      >
        <ul
          class="ant-menu ant-menu-root ant-menu-inline ant-menu-light nav__menu"
          data-menu-list="true"
          role="menu"
          tabindex="0"
        >
          <li
            class="ant-menu-item ant-menu-item-only-child"
            data-menu-id="rc-menu-uuid-test-/profile"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              class="ant-menu-title-content"
            >
              <h3>
                <a
                  class=""
                  href="/profile"
                >
                  Profile
                </a>
              </h3>
            </span>
          </li>
          <li
            class="ant-menu-item ant-menu-item-only-child"
            data-menu-id="rc-menu-uuid-test-/users"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              class="ant-menu-title-content"
            >
              <h3>
                <a
                  class=""
                  href="/users"
                >
                  Users
                </a>
              </h3>
            </span>
          </li>
          <li
            class="ant-menu-item ant-menu-item-only-child"
            data-menu-id="rc-menu-uuid-test-/messages"
            role="menuitem"
            style="padding-left: 24px;"
            tabindex="-1"
          >
            <span
              class="ant-menu-title-content"
            >
              <h3>
                <a
                  class=""
                  href="/messages"
                >
                  Messages
                </a>
              </h3>
            </span>
          </li>
        </ul>
        <div
          aria-hidden="true"
          style="display: none;"
        />
      </nav>
      <main>
        <div
          class="preloader__wrapper"
        >
          <div
            class="ant-spin ant-spin-lg ant-spin-spinning"
          >
            <span
              class="ant-spin-dot ant-spin-dot-spin"
            >
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
              <i
                class="ant-spin-dot-item"
              />
            </span>
          </div>
        </div>
      </main>
      <footer
        class="footer__wrapper"
      >
        Social network ©2021 Created by Balyaev Dmitriy with Ant Design
      </footer>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
